android/app/src/main/java/updated/mysterium/vpn/network/usecase/NodesUseCase.kt

```diff
@@ -0,0 +1,93 @@
+            this.refresh = refresh
+            includeFailed = true
+            serviceType = SERVICE_TYPE
+        }
+        return nodeRepository.proposals(proposalsRequest)
+            .map { NodeEntity.createNodeFromProposal(it) }
+    }
+
+    suspend fun saveAllInitialNodes(nodesList: List<NodeEntity>) {
+        nodeDao.apply {
+            deleteAll()
+            insertAll(nodesList)
+        }
+    }
+
+    suspend fun getAllSavedCountries() = mapNodesToCountriesGroups(nodeDao.getAllNodes())
+
+    private fun mapNodesToCountriesGroups(allNodesList: List<NodeEntity>): List<CountryNodesModel> {
+        val proposalList = createProposalList(allNodesList)
+        return groupListByCountries(proposalList).sortedByDescending { it.proposalList.size }
+    }
+
+    private fun createProposalList(allNodesList: List<NodeEntity>): List<ProposalModel> {
+        val minPricePerByte = allNodesList.minOf { it.pricePerByte }
+        val maxPricePerByte = allNodesList.maxOf { it.pricePerByte }
+        val firstPriceBorder = ((maxPricePerByte - minPricePerByte) / 3) + minPricePerByte
+        val secondPriceBorder = (((maxPricePerByte - minPricePerByte) / 3) * 2) + minPricePerByte
+        return allNodesList.map {
+            ProposalModel.createProposalFromNode(it).apply {
+                calculatePriceLevel(
+                    minPricePerByte,
+                    firstPriceBorder,
+                    secondPriceBorder
+                )
+            }
+        }
+    }
+
+    private fun groupListByCountries(proposalList: List<ProposalModel>): List<CountryNodesModel> {
+        val countryNodesList = mutableListOf<CountryNodesModel>()
+        countryNodesList.add(
+            index = 0,
+            element = CountryNodesModel(
+                countryCode = ALL_COUNTRY_CODE,
+                countryName = "",
+                countryFlagRes = R.drawable.icon_all_countries,
+                proposalList = proposalList
+            )
+        )
+        proposalList.filter { it.countryCode != "" }
+            .forEach { node ->
+                val currentCountry = countryNodesList.find { it.countryCode == node.countryCode }
+                if (currentCountry == null) {
+                    val allNodesByCountry = mutableListOf<ProposalModel>()
+                    proposalList.forEach {
+                        if (it.countryCode == node.countryCode) {
+                            allNodesByCountry.add(it)
+                        }
+                    }
+                    countryNodesList.add(
+                        CountryNodesModel(
+                            countryCode = node.countryCode,
+                            countryName = node.countryName,
+                            proposalList = allNodesByCountry.toList()
+                        )
+                    )
+                }
+            }
+        return countryNodesList.toList()
+    }
+
+    companion object {
```

Comment:
Companion objects should be on top of class before everything else.

Reply:
Fix